### DISCUSSION TITLE
Model compatibility

### DISCUSSION THREAD

**Discussion by TerminalMan** [joeycastillo/Sensor-Watch#27] [General]
Hi,

I hope it's ok to open this topic here?.

I am trying to figure out if this project would in theory be compatible with two other Casio models.

The two models I am specifically interested in are the A168 due to it's electroluminescence backlight, which i was planning to transplant into a A1000 case. Or the G-shock GMW-B5000.

Problem is, it is hard to figure out what kind of display (Backplane config) these models are using.

Due to this project https://github.com/carrotIndustries/pluto I know that the displays are a little finicky, apparently many of the casio watches use displays with 5 backplanes, which apparently is an issue for many uC's.

Based on the Datasheet Page 935, the SAML22 that your project is using also doesn't seem to natively support a 5 backplane config. The pluto project mentioned that running in 8backplane mode would severely hurt the display contrast. Any idea how bad a 6 backplane config would turn out to be?

https://ww1.microchip.com/downloads/aemDocuments/documents/OTH/ProductDocuments/DataSheets/60001465A.pdf

What made you chose the F-91? The conversion cost seems kind of excessive, in relation to the watch price :D

/TM

**Comment by joeycastillo** [comment]
Absolutely OK, thanks for opening up the first discussion! 

> The two models I am specifically interested in are the A168 due to it's electroluminescence backlight, which i was planning to transplant into a A1000 case. Or the G-shock GMW-B5000.

The A168 is almost certainly not compatible. The EL backlight gets its power from two exposed pads on the circuit board, which (based on my limited understanding of EL) would likely have to be boosted to a higher voltage. It's possible the A168 may be using the same LCD, but inside the case the geometry isn't the same, so I doubt the Sensor Watch PCB has everything in the right place. It also lacks the driving circuitry and exposed pads for the backlight.

The GMW-B5000 is definitely incompatible; I can see at a glance the LCD is very different. Having said that, if you have one and want to put some effort into reverse engineering, I offered some tips and tricks [in this talk at Hackaday Remoticon](https://youtu.be/x281ImH68Ik?t=14245).

> Any idea how bad a 6 backplane config would turn out to be?

I do not know, but if I were going to try it, that's likely what I would do. In fact, the Casio Tough Solar (WS220) uses a 6-mux backplane for its main display, everything except the dot matrix up top. [Some notes here](https://twitter.com/josecastillo/status/1463663567882829828); if I were starting a new watch mod today, I'd almost certainly be playing with that one.

> What made you chose the F-91? The conversion cost seems kind of excessive, in relation to the watch price :D

Honestly, it's a lot of the same things in the Pluto issue you linked to: 3 to 3.5 volts is what the SAM L22 supports, so that's where I wanted to be. The CA-53W / CA-506 watches would also fit this criterion and Travis Goodspeed uses this model as a donor watch for [his GoodWatch](https://github.com/travisgoodspeed/goodwatch). But with 35 pins on that LCD, you'd need to move up to a 100-pin chip (the SAM L22 N) to drive them all, and the only way to fit that in a watch case is by moving to a BGA package. 

As for price: I like that the watch itself is cheap enough that you can buy two or three of them: one for wearing, one for testing and one for hacking on. It would be challenging to do the same for a G-Shock, as those things can be pricey. 

Hope this helps!

**Reply by mickael-kerjean** [reply]
wow the Hackaday talk is brillant wizard skills at play, didn't know anything about LCD nor anything that would go around reverse engineering the f91w display to accommodate something else, thank you for unleashing a little bit of that knowledge out there.

**Comment by TerminalMan** [comment]
Thanks for the feedback! The Remoticon talk was great. Esp. the part on reverse engineering the Pinout. 

With theoretically compatible I was referring to the uCs capabilities, a pcb modification will most certainly be necessary. But to figure out if the uC is compatible I guess I have to figure out two things:

- required Voltage levels
- Backplate count

Both doable with a demo unit.

Regarding the EL Backlight, do you have any idea how to pull this off? I am thinking external 5V boost circuit, assuming there is space for it.

The pluto project uses 0.6mm thick PCBs due to space constrains, did you go that route as well?

You mentioned a few uCs in your talk that include a LCD driver. 
- SAM L22
- PIC16LF191 -> sadly a PIC, thus terrible linux/ opensource compiler support :(
- ATmegaXX9
- ATxmega B1/B3
- MSP430F6
- STM32L152 -> What made you decide against this one, pro SAML22?

Did you stumble across any ARM alternatives after your talk, that can boost to 5V?

**Reply by joeycastillo** [reply]
> With theoretically compatible I was referring to the uCs capabilities, a pcb modification will most certainly be necessary.

Ah ok yeah, those two things plus the LCD pin count are definitely the things to look for. Also note you can leave some segment pins floating (and therefore unaddressable) if you don’t have enough pins; depending on the segment map, that may be an acceptable tradeoff. 

> Regarding the EL Backlight, do you have any idea how to pull this off? 

My instinct — and I’m just riffing here, I might be wrong — would be to measure what’s happening on those two EL pads with a multimeter, figure out the voltage and whether it’s DC or AC. Then use a power supply to feed that same voltage back to the EL backlight module, and if it lights, measure how much current it draws. That’ll tell you the requirements for your boost circuit. 

> The pluto project uses 0.6mm thick PCBs due to space constrains, did you go that route as well?

Yep this one is all 0.6 mm PCBs. But I would measure the board in the watch you’re targeting; I forget which one it was, but one watch I took apart had an 0.8mm board. Getting this right is important because a lot of mechanical things like battery contacts and spring connectors depend on the board thickness matching up. 

> STM32L152 -> What made you decide against this one, pro SAML22?

The deciding factor was its similarity to the SAM D21 that folks in my corner of the community know and love. I know how to work with SERCOMs and TC’s on the SAMD series, and the SAM L is similar enough that I feel right at home (which accelerates my development). If you’re familiar with ST’s chips, you may make a different choice. 

> Did you stumble across any ARM alternatives after your talk, that can boost to 5V?

I did not, but one thing you might consider is writing a simple PIC firmware that just uses that PIC16 as your LCD controller; then using a low pin count MCU like the SAM L21 E to control it over SPI. Extra part count is a headache, but if it’s a larger watch and a bigger circuit board you might just have room.

**Comment by TerminalMan** [comment]
> If you’re familiar with ST’s chips, you may make a different choice.

Indeed I am, thanks to ST's brilliant discovery kit marketing :)


Just leaving this here in case you, or someone else needs it in the future~
I started reverse engineering the Casio A168WA and so far it looks good.
 
- The display segment voltages are around ~3V, so it should work with most uC's
- The backlight driver, when not connected, puts out ~10V, there's a small boost circuit on the board with an external inductor.
- The display has 27 Pins, just like the display of the F-91W, I did not find a schematic for the F-91W, thus I don't know if they are compatible.

The PCB Pinout-segment mapping for the A168WA LCD looks like this:
[https://i.imgur.com/gRtN828.png](https://i.imgur.com/gRtN828.png)

**Reply by wormyrocks** [reply]
Just measured backlight on A168, with a battery it's boosting from around -0.4V to 5.33V, at 2KHz with a roughly 50% duty cycle. Same behavior with a fresh battery.

**Comment by joeycastillo** [comment]
Very cool! 

> The display has 27 Pins, just like the display of the F-91W, I did not find a schematic for the F-91W, thus I don't know if they are compatible.

The seemingly identical design, plus the identical placement of the common pins, strongly hint to me that the A168 uses the same LCD glass as the F-91W. On a cursory check against my notes on [the F91W's segment map](https://joeycastillo.github.io/Sensor-Watch-Documentation/segmap.html), every segment I looked at matched up with your diagram.

**Comment by perigoso** [comment]
speaking of model compatibility, and having multiple watches for hacking, does anyone know if the F-91W clones are compatible with the original? importantly in terms of shape and LCD

**Reply by joeycastillo** [reply]
It's _highly unlikely_ that clones will work. Aside from shape, the important thing for this mod is the spacing of the LCD pads and the placement of the LCD common pins. This isn't something the clone manufacturers have any reason to clone precisely, and while only gotten my hands on one clone, its LCD pads are arranged differently. While the Sensor Watch board more or less fits physically, the display is not at all compatible.

This mod will only work with an authentic Casio F-91W.

**Reply by perigoso** [reply]
guess that will have to wait for a custom LCD one day :wink: I might try my luck anyway, I have an authentic, but I was seeing if I could avoid tearing it apart (again)

**Comment by matthewcmead** [comment]
I've been wondering if the W218H-1A with the 3224 module will work with the sensor watch module?  The only reason I'd like to swap the module into the W218H-1A vs the F91W is the 50M water resistance.  Any ideas what kind of water resistance I can expect out of the F91W-1 case in real life practice?

Thanks!

**Reply by wryun** [reply]
Firstly, I can't actually answer your questions, but there are reports on the internet of the F-91W being better than you'd expect in terms of water resistance (i.e. people reporting swimming and showers without issue).

The other thing is, when I was looking for 593 module watches, this site said Casio used to make one which was rated to 50m: http://www.digital-watch.com/DWL/1work/casio-w-39

But when I was searching for ones that were currently made, I could only find: https://www.sensorwatch.net/docs/#need-to-buy-a-watch (none of these are 50m)

**Reply by matthewcmead** [reply]
Thanks, appreciate the details, even if I'm still in search on the W218H-1A.  I might just buy one and take it apart next to the F91W to see.

**Reply by wryun** [reply]
I suspect the LCD Is wired differently if nothing else. You could try putting the 593 module in that case (someone in Discord was doing that with a B640, which required more drastic hackery due to the button positions).

**Reply by Kimm0th** [reply]
I just confirmed the LCD pinout is identical the other day; this is quite doable for anyone keen on a bigger watch and/or screen. Here's the 593 LCD in the 3224 module.
![IMG_20250801_161732_MP](https://github.com/user-attachments/assets/9f0da9e3-0868-4aef-9c6c-485973c1a10f)

Possibilities include grafting the larger LCD into the 593 module (somewhat challenging since it's like 0.3mm wider and you also need to cut away the module chassis to extend the LCD aperture downwards, and then you need a bigger window in your crystal).

Or grafting the SW board into a 3224 module appears more feasible; some minor trimming of the module chassis will allow the PCB to fit, and then you just need to sort out the battery contact and make the buttons work - you'd probably want to swap the light and alarm positions to match the crystal text of the W218H. One nifty way to do this would be to ditch the standard style of contact and use tactile blister buttons, and there's probably also room to inset them enough to make the buttons sit more flush if desired, since the button travel is much shorter on the blister contacts. It should be fairly trivial to squeeze a CR2025 into this case, and the 3224 module chassis is a much nicer two-piece deal which should be less fiddly in this regard (except for the need to adapt the battery contact to the 593/SW position). You can also gain 0.4mm of room for a bigger battery by using the 593 zebra strip, but that will require thinning the module chassis to allow the strip to be clamped... might be worth the hassle if it means you can squeeze a CR2032 in.

Oh yeah, the 3224 module has a pretty decent light spreader:
![IMG_20250801_162124_MP](https://github.com/user-attachments/assets/e3641017-2589-4780-9c02-7a782e8db1bc)
593/SW LED a bit closer to the LCD than this though.

**Reply by jsmcx** [reply]
This is fascinating. I wonder if it would be feasible to modify the SW board so that it would fit both enclosures?

**Reply by Kimm0th** [reply]
> This is fascinating. I wonder if it would be feasible to modify the SW board so that it would fit both enclosures?

Can't imagine how, unless you add bits you can break off so it can be two sizes. Most feasible thing IMO would be a 3D-printed module chassis, using my idea above with the blister buttons, but hard to imagine how you could avoid having to solder them onto the board via some wires

**Reply by bluesrocker22** [reply]
If you only need 50 meters of water resistance and don't have an aesthetic preference for the W218 case, you can use the Casio W59 instead. It has the same PCB as F91, but a different module number due to LCD font differences.

Another partially compatible watch that can only be used with the original LCD (an LCD swap is impossible due to the mask on the glass) is the Casio F-94. The PCB is the same.

@joeycastillo , could you add these models to the compatibility list with my comments?

**Comment by 814d3** [comment]
Found something very interesting and post it here, because of the discussion about the electroluminescence backlight:
https://www.etsy.com/de-en/listing/1448973768/back-light-spreader-for-casio-f-91w-a?ref=reviews
Amazing how simple a very good illumiation can be achived without additional electronics, I really like it.
